contracts/farms/common/RevoFarmBot.sol

```diff
@@ -258,139 +233,17 @@ contract UbeswapFarmBot is ERC20, AccessControl, Pausable {
-
-        // Actually add liquidity
-        liquidityRouter.addLiquidity(
-            address(stakingToken0),
-            address(stakingToken1),
-            _totalAmountToken0,
-            _totalAmountToken1,
-            (_totalAmountToken0 * slippageNumerator) / slippageDenominator,
-            (_totalAmountToken1 * slippageNumerator) / slippageDenominator,
-            address(this),
-            _deadline
-        );
-    }
-
-    /**
-     * The _paths parameter represents a list of paths to use when swapping each rewards token to token0/token1 of the LP.
-     *  Each top-level entry represents a pair of paths for each rewardsToken.
-     *
-     * Example:
-     *  // string token names used in place of addresses for readability
-     *  rewardsTokens = ['cUSD', 'Celo', 'UBE']
-     *  stakingTokens = ['cEUR', 'MOO']
-     *  paths = [
-     *    [ // paths from cUSD to staking tokens
-     *      ['cUSD', 'cEUR'], // order matters here (need first staking token first)
-     *      ['cUSD', 'mcUSD', 'MOO']
-     *    ],
-     *    [ // paths from Celo to staking tokens
-     *      ...
-     *    ],
-     *    [ // paths from UBE to staking tokens
-     *      ...
-     *    ]
-     *  ]
-     *
-     * The _minAmountsOut parameter represents a list of minimum amounts for token0/token1 we expect to receive when swapping
-     *  each rewardsToken. If we do not receive at least this much of token0/token1 for some swap, the transaction will revert.
-     * If a path corresponding to some swap has length < 2, the minimum amount specified for that swap will be ignored.
-     */
-    function compound(
-        address[][2][] memory _paths,
-        uint256[2][] memory _minAmountsOut,
-        uint256 _deadline
-    ) external ensure(_deadline) onlyRole(COMPOUNDER_ROLE) whenNotPaused {
-        require(
-            _paths.length == rewardsTokens.length,
-            "Parameter _paths must have length equal to rewardsTokens"
-        );
-        require(
-            _minAmountsOut.length == rewardsTokens.length,
-            "Parameter _minAmountsOut must have length equal to rewardsTokens"
-        );
-
-        stakingRewards.getReward();
+    // Abstract method for depositing LP into a farm
+    function _deposit(uint256 _lpAmount) internal virtual;
 
-        uint256[] memory _tokenBalances = new uint256[](rewardsTokens.length);
-        for (uint256 i = 0; i < rewardsTokens.length; i++) {
-            _tokenBalances[i] = rewardsTokens[i].balanceOf(address(this));
-        }
+    // Abstract method for withdrawing LP from a farm
+    function _withdraw(uint256 _lpAmount) internal virtual;
 
-        // Perform swaps and add liquidity
-        _addLiquidity(_tokenBalances, _paths, _minAmountsOut, _deadline);
+    // Abstract method for claiming rewards from a farm
+    function _claimRewards() internal virtual;
 
+    // Convenience method for sublcasses
+    function investAllAndSendFees() internal whenNotPaused {
```

Comment:
this function seems out of place to me. No way to claim rewards, just a way to invest them-- kind of weird. I get that you'd want it to be re-usable though. Could it be put into a library instead?